stubs/passlib/passlib/pwd.pyi

```diff
@@ -1,60 +1,91 @@
-from _typeshed import Incomplete
+import random
 from abc import abstractmethod
-from collections.abc import MutableMapping
-from typing import Any
+from collections.abc import Callable, Iterator, MutableMapping, Sequence
+from typing import Any, Literal
+from typing_extensions import TypeAlias
 
 class SequenceGenerator:
-    length: Any
+    length: int | None
     requested_entropy: str
-    rng: Any
+    rng: random.Random
     @property
     @abstractmethod
     def symbol_count(self) -> int: ...
-    def __init__(
-        self, entropy: Incomplete | None = None, length: Incomplete | None = None, rng: Incomplete | None = None, **kwds
-    ) -> None: ...
+    def __init__(self, entropy: int | None = None, length: int | None = None, rng: random.Random | None = None) -> None: ...
     @property
     def entropy_per_symbol(self) -> float: ...
     @property
     def entropy(self) -> float: ...
-    def __next__(self) -> None: ...
-    def __call__(self, returns: Incomplete | None = None): ...
-    def __iter__(self): ...
+    def __next__(self) -> str: ...
+    def __call__(self, returns: None | int | Callable[[Any], Iterator[Any]] = None) -> str | list[str] | Iterator[str]: ...
```

Comment:
This could be improved by using an overload and by using `Self`:

```suggestion
    @overload
    def __call__(self, returns: None = None) -> str: ...
    @overload
    def __call__(self, returns: int) -> list[str]: ...
    @overload
    def __call__(self, returns: Callable[[Any], Iterator[Any]]) -> Self: ...  # "returns" must be the "iter" builtin
```

Also a small nit: We customarily have `None` as last union item.